### PR TITLE
HLEDB v2

### PR DESCRIPTION
**Pull request by LukeUsher**
This is the HLEDB v2 branch rebased onto master, all clean and up to date.

Based on initial testing I am happy to merge this so it can be worked on as part of the main code rather than a separate branch

If all agree, this can be merged and become part of the main codebase already, that way we do not need to keep worrying about it getting too far out-of-sync with master.

### CODE REVIEW

File: src/CxbxKrnl/HLEDataBase/Xapi.OOVPA.inl [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+	REGISTER_OOVPAS(XapiInitProcess, UNPATCHED, 3911, 3950, 4242, 4831, 5028), // obsolete, Too High Level (from 4721's comment)
+	REGISTER_OOVPAS(XapiThreadStartup, UNPATCHED, 3911), // obsolete? (from 4627, 5028, 5558, 5788, 5849's comment) // obsolete (from 4721's comment)
+	REGISTER_OOVPAS(ConvertThreadToFiber, UNPATCHED, 3911),
+	REGISTER_OOVPAS(CreateFiber, UNPATCHED, 3911),
+	REGISTER_OOVPAS(DeleteFiber, UNPATCHED, 3911),
+	REGISTER_OOVPAS(GetExitCodeThread, PATCH, 3911),
+	REGISTER_OOVPAS(GetOverlappedResult, PATCH, 3911),
+	REGISTER_OOVPAS(GetThreadPriority, PATCH, 3911, 5788),
+	REGISTER_OOVPAS(GetTimeZoneInformation, UNPATCHED, 3911),
+	REGISTER_OOVPAS(GetTypeInformation, XREF, 4134), // TODO: Actually introduced in some unknown XDK between 4134 and 4361
```
Comment (jarupxx) [inline src/CxbxKrnl/HLEDataBase/Xapi.OOVPA.inl]: I found the function that you add onto master. No problems.

**Comment by RadWolfie** [comment]
Base on master branch...
Merge Conflict cause:
- GetTypeInformation

File Changes in master before merge: (which affect WIP_HLEDB_v2 branch...)
- HLEDataBase.h
- Xapi.1.0.4134.inl
- Xapi.1.0.4361.inl
- Xapi.1.0.4432.inl
- Xapi.1.0.4627.inl
- Xapi.1.0.4721.inl
- Xapi.1.0.5028.inl
- Xapi.1.0.5233.inl
- Xapi.1.0.5344.inl
- Xapi.1.0.5558.inl
- Xapi.1.0.5788.inl
- Xapi.1.0.5849.inl
- HLEIntercept.cpp
- HLEIntercept.h
(can be viewed via [comparison tool here](https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/compare/0c4d516f232975472cddb4b2ba530af7d1ad1270...Cxbx-Reloaded:e305ff48bbac88d3ec929a96b3c0c9a18850fa86)

I can confirm the rebase on top of master does not cause any corruption.

**Comment by RadWolfie** [comment]
I want to inform testers whose monitoring WIP_HLEDB_v2 improvements to Cxbx-Reloaded. There are chances of regression may occur. I recommend you to "backup" your current HLECache directory, folder, first. Then run Cxbx-Reloaded using v2 database, soon to be merge in master branch, method with clean HLECache directory.

If an issue do regress, it would be the following reason... certain OOVPA isn't lowered enough to detect it or possible false detection from different older revision's function OOVPA. Check with a comparison tool between your backup and the new HLE cache file to verify if database method is the cause of the issue or not. Then proceed file report in game-compatibility repo. We will do our best to get it out of regression fault.

Cheers

**Comment by Sarkie** [comment]
Good work everyone, looks fantastic.

**Comment by PatrickvL** [comment]
I'm extremely happy with the amazing job you guys have pulled off - stellar performance, nothing short of amazing! At times while you were working on this, I was very impressed with how methodical the work was progressing - I feel proud to be small part off such a dedicated, splendid team! Thanks!

**Pull request merged**